ibis/backends/pyspark/compiler.py

```diff
@@ -1,2137 +1,502 @@
-
-        return decorator
-
-    def translate(self, op, *, scope, timecontext, **kwargs):
-        """Translate Ibis expression into a PySpark object.
-
-        All translated expressions are cached within scope. If an expression is
-        found within scope, it's returned. Otherwise, the it's translated and
-        cached for future reference.
-
-        Parameters
-        ----------
-        op
-            An ibis operation.
-        scope
-            dictionary mapping from operation to translated result
-        timecontext
-            time context associated with expr
-        kwargs
-            parameters passed as keyword args
-
-        Returns
-        -------
-        pyspark.sql.DataFrame
-            translated PySpark DataFrame or Column object
-        """
-        # TODO(cpcloud): remove the udf instance checking when going to sqlglot
-        if isinstance(op, ops.ScalarUDF):
-            formatter = compile_scalar_udf
-            result = formatter(self, op, scope=scope, timecontext=timecontext, **kwargs)
-            return result
-        elif (
-            not isinstance(op, ops.ScalarParameter)
-            and (result := scope.get_value(op, timecontext)) is not None
-        ):
-            return result
-        elif (formatter := self._registry.get(type(op))) is not None:
-            result = formatter(self, op, scope=scope, timecontext=timecontext, **kwargs)
-            scope.set_value(op, timecontext, result)
-            return result
-        else:
-            raise com.OperationNotDefinedError(f"No translation rule for {type(op)}")
-
-
-compiles = PySparkExprTranslator.compiles
-
+from ibis.util import gen_name
 
-# TODO(kszucs): there are plenty of repetitions in this file which should be
-# reduced at some point
 
-
-@compiles(ops.DatabaseTable)
-def compile_datasource(t, op, *, timecontext, **kwargs):
-    df = op.source._session.table(op.name)
-    return filter_by_time_context(df, timecontext).alias(op.name)
-
-
-@compiles(ops.SQLQueryResult)
-def compile_sql_query_result(t, op, **kwargs):
-    query, _, client = op.args
-    return client._session.sql(query)
-
-
-def _can_be_replaced_by_column_name(column, table):
-    """Return whether the given `column` can be replaced by its literal name.
-
-    `True` when `column` and `table[column.get_name()]` are semantically equivalent.
-    """
-    # Each check below is necessary to distinguish a pure projection from
-    # other valid selections, such as a mutation that assigns a new column
-    # or changes the value of an existing column.
-    return (
-        isinstance(column, ops.TableColumn)
-        and column.table == table
-        and column.name in table.schema
-        # TODO(kszucs): do we really need this condition?
-        and column == table.to_expr()[column.name].op()
-    )
+@replace(p.RegexExtract | p.RegexReplace | p.RegexSplit | p.RegexSearch)
+def escape_regex_pattern(_):
+    if isinstance(_.pattern, ops.Literal):
```

Comment:
I think this should be done in the literal string translation, since backslashes have the same interpretation there too. Special casing them to regex operations makes string literal behavior thus inconsistent.